### ISSUE TITLE
about the padding C(MXD) -> C(NXD)

### ISSUE THREAD

**Issue by Zero-coder** [MAZiqing/FEDformer#8]
![image](https://user-images.githubusercontent.com/54145971/173165163-e649d61d-5695-46b0-bd80-29fd4a33bea4.png)

```
   def forward(self, q, k, v, mask):#注意细节，key,value,mask都没用上
        # size = [B, L, H, E]-Batch,Length,Height,Element
        B, L, H, E = q.shape
        x = q.permute(0, 2, 3, 1)#why shape?
        # Compute Fourier coefficients
        x_ft = torch.fft.rfft(x, dim=-1) #拿到DFT对时间序列变换的复数值x(0),...,x(n) -> X(0),...,X（n）
        # Perform Fourier neural operations
        out_ft = torch.zeros(B, H, E, L // 2 + 1, device=x.device, dtype=torch.cfloat)
        for wi, i in enumerate(self.index):#创新点：加了一层时序在频域特征的表达
            out_ft[:, :, :, wi] = self.compl_mul1d(x_ft[:, :, :, i], self.weights1[:, :, :, wi])
        # Return to time domain
        x = torch.fft.irfft(out_ft, n=x.size(-1))
        return (x, None) 
```
Dear author:
    Sorry to bother you so many times, one more question about the padding operation,I haven't see the padding process in FEB-module block,could you please mildly explain this.

**Comment by Zero-coder** [comment]
and what's the meaning of [H, E] in size = [B, L, H, E].

**Comment by tianzhou2011** [comment]
Actually, there is no padding part in the code. we just create a zero empty vector and replace some sampling modes, it's like we pad in the beginning.  But in the graph, we draw like we pad afterword.
